lib/hexpm/web/stale.ex

```diff
@@ -0,0 +1,57 @@
+defprotocol Hexpm.Web.Stale do
+  def etag(schema)
+  def last_modified(schema)
+end
+
+defimpl Hexpm.Web.Stale, for: Atom do
+  def etag(nil), do: nil
+
+  # This is not a good solution because we don't know when a missing
+  # association was modified but this is the best we have for now
+  def last_modified(nil), do: ~N[0000-01-01 00:00:00]
```

Comment:
I don't think we can fix this except for not deleting associations and instead marking them as deleted which I don't think I want to do. Open for other ideas...